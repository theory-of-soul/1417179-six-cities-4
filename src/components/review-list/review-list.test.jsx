import React from "react";
import renderer from 'react-test-renderer';
import ReviewList from "./review-list";

const reviewList = [{
  id: 0,
  rating: 2,
  text: `It was a pain. Wi-Fi was soooo low.`,
  date: new Date(`2020-06-10`),
  userName: `Alex`,
  userIcon: `img/avatar-max.jpg`
}];

describe(`ReviewList component snapshot tests`, () => {
  it(`ReviewList with empty list`, () => {
    const tree = renderer
      .create(
          <ReviewList reviewList={[]}/>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`ReviewList with one review`, () => {
    const tree = renderer
      .create(
          <ReviewList reviewList={reviewList}/>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
