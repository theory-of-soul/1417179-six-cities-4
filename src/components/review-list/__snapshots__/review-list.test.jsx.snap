// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList component snapshot tests ReviewList with empty list 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      0
    </span>
  </h2>,
  <ul
    className="reviews__list"
  />,
]
`;

exports[`ReviewList component snapshot tests ReviewList with one review 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Alex
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          It was a pain. Wi-Fi was soooo low.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-06-10"
        >
          June 10, 2020
        </time>
      </div>
    </li>
  </ul>,
]
`;
